frontend/src/components/Layout/Header/SearchCourses/SearchResultItem.tsx

```diff
@@ -30,8 +31,8 @@ const SearchResultItem: React.FC<CourseFound> = ({
     key={course._id}
     to={
       foundInMyCourses
-        ? `${PATHS.myCourses}/${foundInMyCourses}`
-        : `${PATHS.coursesList}/${course._id}`
+        ? transformRoute(PATHS.myCourseDetails, foundInMyCourses)
+        : transformRoute(PATHS.courseDetails, course._id)
```

Comment:
Move please this condition before return statement of this component and define as constant